module/Finna/src/Finna/RecordDriver/SolrEad3.php

```diff
@@ -848,22 +932,31 @@ public function getRelatedPlacesExtended()
      */
     public function getUnitDates()
     {
-        $unitdate = parent::getUnitDate();
-
         $record = $this->getXmlRecord();
-        if (!isset($record->did->unittitle)) {
-            return $unitdate;
-        }
         $result = [];
-        foreach ($record->did->unitdate as $date) {
-            $attr = $date->attributes();
-            if ($desc = $attr->normal ?? null) {
-                $desc = $attr->label ?? null;
+
+        if (isset($record->did->unitdate)) {
+            foreach ($record->did->unitdate as $date) {
+                $attr = $date->attributes();
+                if ($desc = $attr->normal ?? null) {
+                    $desc = $attr->label ?? null;
+                }
+                $date = (string)$date;
+                $result[] = ['data' => (string)$date, 'detail' => (string)$desc];
+            }
+            if ($result) {
+                return $result;
             }
-            $date = (string)$date;
-            $result[] = ['data' => (string)$date, 'detail' => (string)$desc];
         }
-        return $result;
+
+        if (isset($record->did->unitdatestructured->datesingle)) {
```

Comment:
Näitä voi olla useampi, pitäisi varmaan ottaa kaikki, vai?

Reply:
Entä unitdatestructured:n muu sisältö eli vaihtoehtoiset merkintätavat?

Reply:
Lisäsin mukaan kaikki datesingle-elementit.

unitdatestructuredin muita merkintätapoja ei ole käytössä. daterange on mahdollisesti tulossa, mutta laajentaisin vasta kun dataa on saatavilla.